tmail-backend/event-bus-redis/src/main/java/org/apache/james/events/RedisKeyRegistrationHandler.java

```diff
@@ -66,7 +70,12 @@ void start() {
         declarePubSubChannel();
 
         Disposable newSubscription = Mono.from(redisSubscriber.subscribe(registrationChannel.asString()))
-            .thenMany(redisSubscriber.observeChannels())
+            .thenMany(redisSubscriber.observeChannels()
+                .timeout(redisEventBusConfiguration.durationTimeout())
```

Comment:
`.timeout` specs:
```java
	 * Propagate a {@link TimeoutException} as soon as no item is emitted within the
	 * given {@link Duration} from the previous emission (or the subscription for the first item).
```

What if in 10 seconds there is actually no notification (and not Redis outage), then this long-lived Publisher throws the timeout error and disposes?

Reply:
reverted  it